src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/RunWrapper.java

```diff
@@ -125,6 +133,39 @@ public int getNumber() throws AbortException {
         return build().getNumber();
     }
 
+    @Whitelisted
+    public JSONArray getBuildCauses() throws IOException, ClassNotFoundException {
+        return getBuildCauses("hudson.model.Cause");
+    }
+
+    /**
+     * Filters the returned list by the type of <code>Cause</code> class passed as input
+     * ex. <code>getBuildCauess('hudson.model.Cause$UserIdCause')</code> would return only
+     * <code>Cause</code>s of that type
+     *
+     * @param className A string containing the fully qualified name for the class type to filter the result list by
+     * @return a <code>JSONArray</code> of <code>Cause</code>s of the specified type
+     * @throws IOException
+     */
+    @Whitelisted
+    public JSONArray getBuildCauses(String className) throws IOException, ClassNotFoundException {
```

Comment:
Accepting a String classname directly.... eh, not the prettiest thing (oof, that ClassNotFoundException!) but I can't think of another way we might accomplish this.